### ISSUE TITLE
Recommended settings set sourceType to module

### ISSUE THREAD

**Issue by marekdedic** [typescript-eslint/typescript-eslint#118]
The `plugin:@typescript-eslint/recommended` config sets

```json
"parserOptions": {
  "sourceType": "module"
}
```

Might I ask why override the eslint default? I've stumbled upon this while trying to find out why `no-unused-vars` was throwing false positives for functions exported with `/* exported fnName */` - given this override in the recommended config, it makes sense of course, but I had to go through all the settings to find out why it works fine if I enable only the rule and throws false positives if I extend the whole recommended config...

I get that modules are The Right Way™ now, but not everybody uses them - overriding the eslint default seems like a poor choice to me, because it silently changes the behaviour of eslint without the user knowing why if they don't actually go into the recommended config file.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.0.0` |
| `@typescript-eslint/parser`        | `1.0.0` |
| `TypeScript`                       | `3.2.4` |
| `ESLint`                           | `5.12.1` |
| `node`                             | `11.7.0` |
| `npm`                              | `6.6.0` |

**Comment by bradzacher** [comment]
When we put the recommended config up for debate we erred on the side of what's the most common / correct use case - creating minimal config for the majority with overrides for the minority.

The majority use case is for `sourceType: module`, so if we don't set that in our recommended settings, then the majority of recommended users would have to manually add that setting.

It's certainly up for discussion, if enough people feel strongly against this setting, we can remove it.

**Comment by platinumazure** [comment]
We could also export multiple configs that are "recommended", one for scripts and one for modules (or just one with the module override and one without). As it is, having one recommended config be opinionated feels a little weird (to me).

**Comment by bradzacher** [comment]
What's recommended is always going to be subjective to some degree 😄 

The recommended ruleset we eventually landed on was based on two things:
1) amalgamating several existing configs from the community
2) what the TS team does within their documentation

https://github.com/bradzacher/eslint-plugin-typescript/issues/144

Definitely not opposed to exporting *more* configs though.
I think there is real value in having a number of default configs to make adoption easier.

**Comment by GenaBitu** [comment]
Hi, thanks for the responses.

yes, this is definitely subjective. My point of view is that I thought of the recommended settings as a parallel to the recommended settings from eslint itself - just a set of rules to be enabled, that most people would agree are useful, with some of the more obscure onea left out. Therefore, I was also surprised that it set the parser, but that's obviously useful for everyone using the plugin.

**Comment by bradzacher** [comment]
you're not blind - It's not documented because it's a rare case that hasn't been considered before.
